File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        p = 1
        num = str(n)
        for ch in num:
            p *= int(ch)
        if p % t == 0:
            return n
        return self.smallestNumber(n+1,t)

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        string num = to_string(n);
        int p = 1;
        for(char ch : num){
            p *= (ch - '0');
        }
        if(p % t == 0) return n;
        return smallestNumber(n+1,t);
    }
};